modules/transport-netty4/src/test/java/org/elasticsearch/rest/discovery/Zen2RestApiIT.java

```diff
@@ -0,0 +1,152 @@
+            .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), 2)
+            .put(ClusterBootstrapService.INITIAL_MASTER_NODE_COUNT_SETTING.getKey(), 2)
+            .put(DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.getKey(), "5s")
+            .build();
+    }
+
+    @Override
+    protected boolean addMockHttpTransport() {
+        return false; // enable http
+    }
+
+    public void testAddAndClearVotingTombstones() throws Exception {
+        final int nodeCount = 2;
+        final List<String> nodes = internalCluster().startNodes(nodeCount);
+        createIndex("test",
+            Settings.builder()
+                .put(UnassignedInfo.INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.getKey(), TimeValue.ZERO) // assign shards
+                .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, nodeCount) // causes rebalancing
+                .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 1)
+                .build());
+        ensureGreen("test");
+
+        RestClient restClient = getRestClient();
+
+        internalCluster().rollingRestart(new InternalTestCluster.RestartCallback() {
+            @Override
+            public void doAfterNodes(int n, Client client) throws IOException {
+                ensureGreen("test");
+                Response response =
+                    restClient.performRequest(new Request("POST", "/_cluster/withdrawn_votes/" + internalCluster().getNodeNames()[n]));
+                assertThat(response.getStatusLine().getStatusCode(), is(200));
+            }
+
+            @Override
+            public Settings onNodeStopped(String nodeName) throws IOException {
+                String viaNode = randomValueOtherThan(nodeName, () -> randomFrom(nodes));
+
+                List<Node> allNodes = restClient.getNodes();
+                try {
+                    restClient.setNodes(
+                        Collections.singletonList(
+                            new Node(
+                                HttpHost.create(
+                                    internalCluster().getInstance(HttpServerTransport.class, viaNode)
+                                        .boundAddress().publishAddress().toString()
+                                )
+                            )
+                        )
+                    );
+                    Response deleteResponse = restClient.performRequest(new Request("DELETE", "/_cluster/withdrawn_votes"));
+                    assertThat(deleteResponse.getStatusLine().getStatusCode(), is(200));
+
+                    final ClusterHealthRequestBuilder clusterHealthRequestBuilder = client(viaNode).admin().cluster().prepareHealth()
+                        .setWaitForEvents(Priority.LANGUID)
+                        .setWaitForNodes(Integer.toString(nodeCount - 1))
+                        .setTimeout(TimeValue.timeValueSeconds(30L));
+
+                    clusterHealthRequestBuilder.setWaitForYellowStatus();
```

Comment:
This can be in the chain of `.set()` methods above, and the temporary `clusterHealthRequestBuilder` can probably be inlined.